pkgs/applications/networking/firewalld/default.nix

```diff
@@ -0,0 +1,110 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoreconfHook
+, docbook_xml_dtd_42
+, docbook_xsl
+, glib
+, gobject-introspection
+, gtk3
+, intltool
+, libnotify
+, libxml2
+, libxslt
+, networkmanager-applet
+, pkg-config
+, python3
+, wrapGAppsHook
+, withGui ? false
+}:
+
+stdenv.mkDerivation rec {
+  pname = "firewalld";
+  version = "1.1.1";
+
+  src = fetchFromGitHub {
+    owner = "firewalld";
+    repo = "firewalld";
+    rev = "v${version}";
+    sha256 = "sha256-w8TbovIhOhJAUZWbKdBd/+db8Hro/ttlxWZDcrCXX4Q=";
+  };
+
+  postPatch = ''
+    for file in doc/xml/transform-{html,man}.xsl.in; do
+      substituteInPlace $file \
+        --replace "http://docbook.sourceforge.net/release/xsl/current" "${docbook_xsl}/xml/xsl/docbook"
+    done
+
+    for f in doc/xml/*.xml* ; do
+      substituteInPlace $f \
+        --replace "http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd" "${docbook_xml_dtd_42}/xml/dtd/docbook/docbookx.dtd"
+    done
+
+    substituteInPlace src/firewall/config/__init__.py.in \
+      --replace "/usr/share" "$out/share"
+
+    for file in config/firewall-{applet,config}.desktop.in; do
+      substituteInPlace $file \
+        --replace "/usr/bin/" "$out/bin/"
+    done
+
+    patchShebangs --host src/gtk3_chooserbutton.py src/gtk3_niceexpander.py src/tests/python/
```

Comment:
Why do what I presume are custom widgets (i.e. python modules) need patched shebangs? And should not tests rather have `--build` shebangs?

Reply:
🤔 I should probably add them to wrapPythonPrograms and try to run tests as part of building the package.

Reply:
The tests actually get installed into $out. I think they are runtime tests to check if your firewalld is working correct.